libs/application/templates/new-primary-school/src/lib/dataSchema.ts

```diff
@@ -133,29 +133,25 @@ export const dataSchema = z.object({
       languageEnvironment: z.string(),
       signLanguage: z.enum([YES, NO]),
       interpreter: z.string().optional(),
-      language1: z.string().optional().nullable(),
-      language2: z.string().optional().nullable(),
+      selectedLanguages: z.array(z.object({ code: z.string() })).optional(),
```

Comment:
I can't proceed to next page if select `ICELANDIC_AND_OTHER` / `ONLY_OTHER_THAN_ICELANDIC` and then change to `ONLY_ICELANDIC` without selecting any languages. 

Error:
```javascript
{
    "selectedLanguages": [
        {
            "code": "Ógilt gildi."
        },
        {
            "code": "Ógilt gildi."
        }
    ]
}
``` 

Maybe need to change `code` to optional and nullable?
```suggestion
      selectedLanguages: z.array(z.object({ code: z.string().optional().nullable() })).optional(),
```

Reply:
Fixed